aspnet/fundamentals/logging.rst

```diff
@@ -1,8 +1,202 @@
-.. include:: /../common/stub-topic.txt
+Logging
+=======
 
-|stub-icon| Logging
-===================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+ASP.NET 5 has built-in support for logging, and allows developers to easily leverage their preferred logging framework's functionality as well. Implementing logging in your application requires a minimal amount of setup code. Once this is in place, logging can be added wherever it is desired.
+
+In this article:
+	- `Implementing Logging in your Application`_
+	- `Configuring Logging in your Application`_
+	
+`View or download sample from GitHub <https://github.com/aspnet/Home/tree/dev/samples>`_.
+
+Implementing Logging in your Application
+----------------------------------------
+
+Adding logging to your application is done by requesting either an ``ILoggerFactory`` or an ``ILogger<T>`` via :doc:`dependency-injection`. If an ``ILoggerFactory`` is requested, a logger must be created using its ``CreateLogger`` method. The following example shows how to do this within the ``Configure`` method in the ``Startup`` class:
+
+.. literalinclude:: logging/sample/src/LoggingSample/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 17-27
+	:dedent: 8
+	:emphasize-lines: 2,7-8
+
+When a logger is created, a category name or source must be provided. By convention this string is hierarchical, with categories separated by dot (``.``) characters. Some logging providers have filtering support that leverages this convention, making it easier to locate logging output of interest. In the above example, the logging is configured to use the built-in `ConsoleLogger <https://github.com/aspnet/Logging/blob/1.0.0-beta6/src/Microsoft.Framework.Logging.Console/ConsoleLogger.cs>`_ (see `Configuring Logging in your Application`_ below). To see the console logger in action, run the sample application using the ``web`` command, and make a request to configured URL (``localhost:5000``). You should see output similar to the following:
+
+.. image:: logging/_static/console-logger-output.png
+
+You may see more than one log statement per web request you make in your browser, since most browsers will make multiple requests (i.e. for the favicon file) when attempting to load a page. Note that the console logger displayed the log level (``info`` in the image above) followed by the category (``[LoggingSample.Startup]``), and then the message that was logged.
+
+Logging Verbosity Levels
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When adding logging statements to your application, you must specify a `LogLevel <https://github.com/aspnet/Logging/blob/1.0.0-beta6/src/Microsoft.Framework.Logging.Abstractions/LogLevel.cs>`_. The LogLevel allows you to control the verbosity of the logging output from your application, as well as the ability to pipe different kinds of log messages to different loggers. For example, you may wish to log debug messages to a local file, but log errors to the machine's event log or a database.
+
```

Comment:
Should emphasize here (or another more appropriate place) the capability to dial up and down info from the framework. We're super minimal by default, but this can be a valuable tool.